cache/remotecache/export.go

```diff
@@ -74,21 +132,23 @@ func (ce *contentCacheExporter) Finalize(ctx context.Context) (map[string]string
 		return nil, err
 	}
 
-	// own type because oci type can't be pushed and docker type doesn't have annotations
-	type manifestList struct {
-		specs.Versioned
-
-		MediaType string `json:"mediaType,omitempty"`
-
-		// Manifests references platform specific manifests.
-		Manifests []ocispecs.Descriptor `json:"manifests"`
+	cache := ExportableCache{}
```

Comment:
iiuc the public functions `SetMediaType` and `SetSchemaVersion` are not needed. You can just make a constructor `NewExportableCache(type)` and everything is set up internally in that function.

For the `oci` check, this can be before `NewExportableCache` is called.